container-search/src/main/java/com/yahoo/search/result/FeatureData.java

```diff
@@ -74,10 +74,14 @@ public StringBuilder writeJson(StringBuilder target) {
      *                                  (that is, if it is a tensor with nonzero rank)
      */
     public Double getDouble(String featureName) {
-        if (decodedDoubles != null && decodedDoubles.containsKey(featureName))
-            return decodedDoubles.get(featureName);
-        Double value = decodeDouble(featureName);
-        if (decodedDoubles == null)
+        Double value = null;
+        if (decodedDoubles != null)
+            value = decodedDoubles.get(featureName);
+        if (value != null)
+            return value;
+
+        value = decodeDouble(featureName);
+        if (value != null && decodedDoubles == null)
```

Comment:
this must still be just:
```
    if (decodedDoubles == null)
            decodedDoubles = new HashMap<>();	
```
same with decodedTensors below.
but since in all common cases you will always create it, I would suggest doing that first in the method.  That should make the code easier to read.

Reply:
Good suggestion, I'll do that.